sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -757,7 +758,10 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(WAREHOUSE_PATH)).toString
+  def warehousePath: String = {
+    val path = new Path(getConf(WAREHOUSE_PATH))
+    FileSystem.get(path.toUri, new Configuration()).makeQualified(path).toString
```

Comment:
`Utils.resolveURI` creates an `java.net.URI` from a String path, and returns the URI if it has schema. Apparently, it treats Windows paths as having a schema and breaks down with back slashes. `Utils.resolveURI` returns a different results comparing to hadoop `Path.toUri`:
````
scala> (new URI("/my")).getScheme()
res22: String = null

scala> (new URI("c:/my")).getScheme()
res23: String = c

scala> (new URI("c:\\my")).getScheme()
java.net.URISyntaxException: Illegal character
````
hadoop Path adds a slash to the path:
````
scala> new Path("c:/my").toUri
res25: java.net.URI = /c:/my
````